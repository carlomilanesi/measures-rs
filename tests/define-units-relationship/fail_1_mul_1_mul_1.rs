measures::define_measure_types! {
    exact,
    scalar_properties [ ]
    vector_properties [
        P1 [
            U1 {
                suffix: " u1",
            }
        ]
        P2 [
            U2 {
                suffix: " u2",
            }
        ]
        P3 [
            U3 {
                suffix: " u3",
            }
        ]
    ]
    angle_measurement_units [ ]
    relationships [
        U1 1 * U2 1 * U3 1,
    ]
}

fn main() {}
