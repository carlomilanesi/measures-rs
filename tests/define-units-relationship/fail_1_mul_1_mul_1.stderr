error: no rules expected `*`
  --> tests/define-units-relationship/fail_1_mul_1_mul_1.rs:23:14
   |
23 |         U1 1 * U2 1 * U3 1,
   |              ^ no rules expected this token in macro call
   |
note: while trying to match `==`
  --> src/define_measure_types.rs
   |
   |                 $unit1:ident $dim1:tt == $unit2:ident $dim2:tt $op:tt $unit3:ident $dim3:tt $(,)?
   |                                       ^^
